11-Backbonejs/Backbone-Views.md

```diff
@@ -0,0 +1,149 @@
+
+![Backbone Views](images/backbone-views.png)
+
+You can see a deployed version of this app [here:](http://cheezitman.github.io/tasklist-js)
+
+## Creating A TaskView
+
+![Application Overview](images/Ada-TaskList-JS.png)
+
+A TaskView will handle drawing an individual task item and responding to events that concern just that task.  To start we create a file in `app/views/task_view.js`  Each TaskView will only be concerned with rendering a specific task and dealing with events on that task.  We've indicated it in red in the image above.  
+
+
+```JavaScript
+import Backbone from 'backbone';
+import _ from 'underscore';
+import $ from 'jquery';
+import Task from '../models/task.js';
+
+var TaskView = Backbone.View.extend({
+  initialize: function(params) {
+    this.template = params.template;
+  },
+  render: function() {
+    var compiledTemplate = this.template(this.model.toJSON());
+    this.$el.html(compiledTemplate);
+    return this;
+  }
+});
+
+export default TaskView;
+```
+
+Just like Models and Collections a view extends `Backbone.View`.  This model has 4 important properties, `initialize`, `render` `el`, & `model`.
+
+|   Property	|   Description	|
+|---	|---	|
+|   `initialize`	|   A function, like in a Ruby class, called immediately when a new Backbone object is created, View, Model or Collection. 	|
+|   `el`	|   An HTML DOM element that, by default, is an empty `div`.  We use `el` to insert our view into the page when it is rendered.  	|
+|   `$el`	|   A jQuery selection of `el`, and you can use jQuery functions on it.  With `$el` you can run jQuery selections for HTML elements inside, and only inside the view.  For example, `$el.find('button.delete')` is jQuery selection of button with the class `delete` inside of `el` (the view). 	In other words `myView.$el` and `$(myView.el)` are equivalent.  |
+|   `model`	|   The Backbone model which provides the data for the view.  The view's `model` can be a Backbone Model or Collection.  Each View should, in general, have **one** `model`.    	|
+|   `render`	|   A function called to draw (or redraw) the view.  By convention the render function always returns `this` so that it can be chained with other methods.  |
+
+
+## Adding our view to `app.js`
+
+We can update our renderList method to use the new view to draw each element.  Instead of calling the render method we created in `app.js` we create a new TaskView with the given model & template.
+
+Then we render the template and append the resulting `$el` to the DOM.  
+
+```javascript
+// app.js
+// imports etc...
+import TaskView from './views/task_view.js';
+
+var renderList = function(taskList) {
+  // Clear the unordered list
+  $('.todo-items').empty();
+  
+  // Iterate through the list rendering each Task
+  taskList.each(function(task) {
+  
+    // Create a new TaskView with the model & template
+    var taskView = new TaskView({
```

Comment:
I'd had a section at the end about optimizations, but we took it out as it seemed too much too early.